var/spack/repos/builtin/packages/qgis/package.py

```diff
@@ -185,6 +187,7 @@ def cmake_args(self):
                 "-DLIBZIP_INCLUDE_DIR=" + self.spec["libzip"].prefix.include,
                 "-DLIBZIP_CONF_INCLUDE_DIR=" + self.spec["libzip"].prefix.lib.libzip.include,
                 "-DGDAL_CONFIG_PREFER_PATH=" + self.spec["gdal"].prefix.bin,
+                "-DGDAL_LIBRARY=" + self.spec["gdal"].prefix.lib+'/libgdal.so',
```

Comment:
```suggestion
                "-DGDAL_LIBRARY=" + self.spec["gdal"].libs[0],
```
This will also work on macOS